### PR TITLE
[IMP] web_tour: add debug mode in start_tour

### PR DESCRIPTION
**Pull request by pipu-odoo**
In this commit, we add the possibility of starting a tour in debug=True
mode. In this mode, the browser opens in a large window with a devtools
and a breakpoint is triggered at the start of the test.
This allows the devtools to be configured before the tour runs.

We also added features for steps :
step.break => the tour stops at the start of the step by a debugger.
	This makes it easier to debug a specific step.
step.pause => the round stops at the end of the step. This allows you
	to manipulate the dom and check the next step before it processes.

We have also improved the error messages in order to target where
it is located in the step.

### CODE REVIEW

**Review by Arcasias — CHANGES_REQUESTED** [review]

File: addons/web_tour/static/src/tour_service/tour_compilers.js [hunk 1]
```diff
@@ -288,15 +341,12 @@ export function compileStepAuto(stepIndex, step, options) {
...
-                    browser.clearTimeout(tourTimeout);
-                    tourTimeout = browser.setTimeout(() => {
-                        // The logged text shows the relative position of the failed step.
-                        // Useful for finding the failed step.
-                        console.warn(describeFailedStepDetailed(step, tour));
-                        // console.error notifies the test runner that the tour failed.
-                        console.error(describeFailedStepSimple(step, tour));
-                    }, (step.timeout || 10000) + stepDelay);
+                browser.clearTimeout(tourTimeout);
+                if (!keepWatchBrowser && !step.freeze) {
```
[the remark this answers is not in the document]
Comment (pipu-odoo) [reply addons/web_tour/static/src/tour_service/tour_compilers.js]: cofee ?

**Review by Arcasias — APPROVED** [review]
@robodoo r+

**Comment by fja-odoo** [comment]
@pipu-odoo Hello, just checking if you are planning to add some info about that in the documentation ?
https://www.odoo.com/documentation/master/developer/reference/backend/testing.html#debugging-tips

**Comment by pipu-odoo** [comment]
@fja-odoo Here is the documentation https://github.com/odoo/documentation/pull/7808

**Comment by pipu-odoo** [comment]
this PR was replaced by https://github.com/odoo/odoo/pull/155082.

**Pull request closed without merging**